protocols/rendezvous/src/client.rs

```diff
@@ -76,19 +124,38 @@ impl Behaviour {
     /// External addresses are either manually added via [`libp2p_swarm::Swarm::add_external_address`] or reported
     /// by other [`NetworkBehaviour`]s via [`ToSwarm::ExternalAddrConfirmed`].
     pub fn register(&mut self, namespace: Namespace, rendezvous_node: PeerId, ttl: Option<Ttl>) {
-        self.pending_register_requests
-            .push((namespace, rendezvous_node, ttl));
+        let external_addresses = self.external_addresses.iter().cloned().collect::<Vec<_>>();
+        if external_addresses.is_empty() {
+            self.error_events
+                .push_back(Event::RegisterFailed(RegisterError::NoExternalAddresses));
+
+            return;
+        }
```

Comment:
I think it would make sense to return this error directly from this function.

This is only possible now because we no longer depend on `PollParameters`.

However, I think it is better to have this PR be a pure refactoring so we can do this in a follow-up!

Reply:
So, do we leave it as is?

Reply:
Yes, leave it as is in this PR and let's record a follow-up issue!

Reply:
ok